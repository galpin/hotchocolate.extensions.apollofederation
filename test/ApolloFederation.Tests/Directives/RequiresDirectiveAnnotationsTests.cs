using System.Threading.Tasks;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Xunit;
using static HotChocolate.Extensions.ApolloFederation.Test;

#nullable disable

namespace HotChocolate.Extensions.ApolloFederation.Directives;

public class RequiresDirectiveAnnotationsTests
{
    [Fact]
    public async Task Ctor_correctly_configures_directive()
    {
        var schema = await BuildSchemaAsync(x => x.AddQueryType<Query<Review>>());

        var sut = schema.GetType<ObjectType>("Review");

        Assert.Collection(
            sut.Fields["product"].Directives,
            x => AssertEx.Directive(x, "requires", ("fields", "\"id\"")));
    }

    public class Query<T>
    {
        public T Get(int id)
        {
            return default;
        }
    }

    public class Review
    {
        [GraphQLKey]
        public int Id { get; set; }

        [GraphQLRequires("id")]
        public Product Product { get; set; }
    }

    public class Product
    {
        public string Name { get; set; }
    }
}