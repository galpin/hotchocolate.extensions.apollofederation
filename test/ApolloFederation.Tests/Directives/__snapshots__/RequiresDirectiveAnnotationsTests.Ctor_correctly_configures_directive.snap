{
  "data": {
    "_service": {
      "sdl": "type Product {\n  upc: String!\n}\n\ntype Review @key(fields: \u0022id\u0022) {\n  id: Int!\n  product: Product! @requires(fields: \u0022id\u0022)\n}"
    }
  }
}
